gui/wxpython/history/tree.py

```diff
@@ -209,97 +230,113 @@ def _popupMenuEmpty(self):
         self.PopupMenu(menu)
         menu.Destroy()
 
-    def _timestampToDay(self, timestamp=None):
+    def _timestampToDay(self, datetime_timestamp=None):
         """
-        Convert timestamp to datetime.date object with time set to midnight.
-        :param str timestamp: Timestamp as a string in ISO format.
+        Convert a timestamp to datetime.date object with time set to midnight.
+
+        :param datetime.datetime datetime_timestamp: Command timestamp
         :return datetime.date day_midnight: midnight of a day.
         """
-        if not timestamp:
+        if not datetime_timestamp:
             return OLD_DATE
 
-        timestamp_datetime = datetime.datetime.fromisoformat(timestamp)
         day_midnight = datetime.datetime(
-            timestamp_datetime.year, timestamp_datetime.month, timestamp_datetime.day
+            datetime_timestamp.year, datetime_timestamp.month, datetime_timestamp.day
         ).date()
 
         return day_midnight
 
+    def _timestampToISO(self, datetime_timestamp=None):
+        """
+        Convert a datetime timestamp to ISO format.
+
+        :param str datetime_timestamp: datetime.datetime object .
+        :return: Command timestamp in ISO format or None if datetime_timestamp is None.
+        """
+        return (
+            datetime.datetime.isoformat(datetime_timestamp)
+            if datetime_timestamp
+            else None
+        )
+
+    def _timestampToDatetime(self, iso_timestamp=None):
+        """
+        Convert an ISO format timestamp string to a datetime object.
+
+        :param str iso_timestamp: Command timestamp in ISO format.
+        :return: datetime.datetime object or None if iso_timestamp is None.
+        """
+        return datetime.datetime.fromisoformat(iso_timestamp) if iso_timestamp else None
+
+    def _reloadNode(self, node):
+        """Reload the model of a specific node"""
+        self._model.SortChildren(node)
+        self.SetModel(copy.deepcopy(self._model))
```

Comment:
It may work, but this line looks strange to me...

Reply:
you mean the whole method `_reloadNode`?

Reply:
But otherwise it looks like it works nicely!

Reply:
Well, the behavior is a bit strange here.. First I assumed that it should work without SetModel line, just:
```

    self._model.SortChildren(node)
    self._orig_model = copy.deepcopy(self._model)
```
    
similarly as it is in a datacatalog.
But in our case, the history tree is not loaded - just empty panel is shown .. So I tried another variant:

```
def _reloadNode(self, node):
    """Reload the model of a specific node"""
    self._model.SortChildren(node)
    self._orig_model = copy.deepcopy(self._model)
    if node != self._model.root:
        self.RefreshNode(node)
```
but the result was the same, again no error and the panel is empty.

What works is this:

```
   def _reloadNode(self, node):
        """Reload the model of a specific node"""
        self._model.SortChildren(node)
        self._orig_model = copy.deepcopy(self._model)
        self.SetModel(self._model)
```
        
which looks a bit better than the original method but no glory.

Reply:
Well, the behavior is a bit strange here.. First I assumed that it should work without SetModel line, just:
```

    self._model.SortChildren(node)
    self._orig_model = copy.deepcopy(self._model)
```
    
similarly as it is in a datacatalog.
But in our case, the history tree is not loaded - just empty panel is shown .. So I tried another variant:

```
def _reloadNode(self, node):
    """Reload the model of a specific node"""
    self._model.SortChildren(node)
    self._orig_model = copy.deepcopy(self._model)
    if node != self._model.root:
        self.RefreshNode(node)
```
but the result was the same, again no error and the panel is empty.

What works is this:

```
   def _reloadNode(self, node):
        """Reload the model of a specific node"""
        self._model.SortChildren(node)
        self._orig_model = copy.deepcopy(self._model)
        self.SetModel(self._model)
```
        
which looks a bit better than the original method but no glory.

Reply:
RefreshItems helped .. see https://github.com/OSGeo/grass/pull/3896/commits/a72e5522c3ff12da5497b1dd7af103ce96d62281